packages/core/src/factory.ts

```diff
@@ -18,31 +18,9 @@ export function factory<Resource, Input, Output, Config extends IPrismConfig>(
       const inputValidations: IPrismDiagnostic[] = [];
 
       return pipe(
-        components.route({ resources, input, config }),
-        Either.fold(
-          error => {
-            // rethrow error we if we're attempting to mock
-            if (config.mock) {
-              return TaskEither.left(error);
-            }
-
-            const { message, name, status } = error as ProblemJsonError;
-            // otherwise let's just stack it on the inputValidations
-            // when someone simply wants to hit an URL, don't block them
-            inputValidations.push({
-              message,
-              source: name,
-              code: status,
-              severity: DiagnosticSeverity.Warning,
-            });
-
-            return TaskEither.right<Error, Resource | undefined>(undefined);
-          },
-          value => TaskEither.right(value),
-        ),
+        TaskEither.fromEither(components.route({ resources, input })),
         TaskEither.chain(resource => {
-          // validate input
-          if (config.validateRequest && resource && components.validateInput) {
+          if (config.validateRequest && resource) {
             inputValidations.push(
```

Comment:
if we're doing this refactor, let's not force the monad to go outside its `default minimal context` - https://drboolean.gitbooks.io/mostly-adequate-guide-old/content/ch9.html. So if we need `inputValidations`, let's pass it through the chain or use `Reader` monad. And also, most importantly, let's not `.push`, especially inside of a monad

Reply:
I do not think it's going to happen in this PR, unfortunately.

Reply:
Ok, sure :+1: